lib/service/service.go

```diff
@@ -1183,6 +1224,43 @@ func NewTeleport(cfg *servicecfg.Config) (*TeleportProcess, error) {
 	return process, nil
 }
 
+func (process *TeleportProcess) updateDownstreamPingHandler(cfg *servicecfg.Config) {
+	upgraderKind := os.Getenv(externalUpgraderEnv)
+
+	process.inventoryHandleMu.Lock()
+	defer process.inventoryHandleMu.Unlock()
+
+	oldInventoryHandle := process.inventoryHandle
+
+	instanceRoles := append(process.getInstanceRoles(), process.getDynamicInstanceRoles()...)
+
+	process.inventoryHandle = inventory.NewDownstreamHandle(process.makeInventoryControlStreamWhenReady, proto.UpstreamInventoryHello{
```

Comment:
There are other places in this file that access this field without holding `inventoryHandleMu`. The `inventoryHandle` is also passed to `regular.New()` which may end up holding an old `inventoryHandle`. Is replacing `inventoryHandle` the right thing to do here, or would it be better to tell the existing one of the change of roles?